src/react-sortable-tree.js

```diff
@@ -230,7 +230,7 @@ class ReactSortableTree extends Component {
     // it means that the drag was canceled or the dragSource dropped
     // elsewhere, and we should reset the state of this tree
     if (!monitor.isDragging() && this.state.draggingTreeData) {
-      this.endDrag();
+      setTimeout(() => {this.endDrag()});
```

Comment:
please describe what's to do in this statement. And write a test case to make this PR available to merge and realese

Reply:
This should be high priority

Reply:
I can't figure out a way to write a test case for this, but here's an attempt to describe what's happening:

    // If the drag ends and the tree is still in a mid-drag state, it
    // means that the drag was canceled or the dragSource dropped
    // elsewhere, and we should reset the state of this tree.
    //
    // However, right now react-dnd is still processing its events, and
    // if we unmount the mid-drag React components immediately (by calling
    // this.endDrag()), that processing will run into errors (see 
    // https://github.com/react-dnd/react-dnd/issues/1368 for details). 
    // So let react-dnd process its events completely (which happens synchronously)
    // before we call endDrag).